File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        dic={}
        for w,l in matches:
            dic[w]=1
            dic[l]=1
        loser=[i[1] for i in matches]
        for i in loser:
            dic[i]=0
        x=Counter(loser)
        res=[[k for k,v in dic.items() if v==1],[k for k,v in x.items() if v==1]]
        res[0].sort()
        res[1].sort()
        return res

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int,pair<int,int>>mp;
        vector<vector<int>>v(2);
        for(auto it:matches){
            mp[it[0]].first++;
            mp[it[1]].second++;
        }
        
        for(auto it:mp){
            if(it.second.second==0){
                v[0].push_back(it.first);
            }
            else if(it.second.second ==1){
                v[1].push_back(it.first);
            }
        }
        sort(v[0].begin(),v[0].end());
        sort(v[1].begin(),v[1].end());
        return v;
    }
};